Firestore.js

```diff
@@ -26,92 +27,99 @@ var Firestore = function (email, key, projectId) {
+  this.getDocuments = function (path) {
+    return this.query(path).execute()
   }
 
   /**
-   * Update/patch a document at the given path with new fields.
+   * Get a list of all IDs of the documents in a path
    *
-   * @param {string} path the path of the document to update
-   * @param {object} fields the document's new fields
-   * @return {object} the Document object written to Firestore
+   * @param {string} path the path to the collection
+   * @return {object} an array of IDs of the documents in the collection
    */
-  this.updateDocument = function (path, fields) {
-    return updateDocument_(path, fields, authToken, projectId)
+  this.getDocumentIds = function (path) {
+    const request = new FirestoreRequest_(baseUrl, authToken)
+    return getDocumentIds_(path, request)
   }
 
   /**
-   * Get a list of all documents in a collection.
+   * Create a document with the given fields and an auto-generated ID.
    *
-   * @param {string} pathToCollection the path to the collection
-   * @return {object} an array of the documents in the collection
+   * @param {string} path the path where the document will be written
+   * @param {object} fields the document's fields
+   * @return {object} the Document object written to Firestore
    */
-  this.getDocuments = function (pathToCollection) {
-    return getDocuments_(pathToCollection, authToken, projectId)
+  this.createDocument = function (path, fields) {
+    const request = new FirestoreRequest_(baseUrl, authToken)
+    return createDocument_(path, fields, request)
   }
 
   /**
-   * Get a document.
+   * Update/patch a document at the given path with new fields.
    *
-   * @param {string} path the path to the document
-   * @return {object} the document object
+   * @param {string} path the path of the document to update.
+   *                      If document name not provided, a random ID will be generated.
+   * @param {object} fields the document's new fields
+   * @return {object} the Document object written to Firestore
    */
-  this.getDocument = function (path) {
-    return getDocument_(path, authToken, projectId)
+  this.updateDocument = function (path, fields) {
+    const request = new FirestoreRequest_(baseUrl, authToken)
+    return updateDocument_(path, fields, request)
   }
 
   /**
-   * Get a list of all IDs of the documents in a collection.
+   * Update/patch a document at the given path with new fields.
    *
-   * @param {string} pathToCollection the path to the collection
-   * @return {object} an array of IDs of the documents in the collection
+   * @param {string} path the path of the document to update
+   * @param {object} fields the document's new fields
+   * @return {object} the Document object written to Firestore
    */
-  this.getDocumentIds = function (pathToCollection) {
-    return getDocumentIds_(pathToCollection, authToken, projectId)
+  this.updateDocument = function (path, fields) {
```

Comment:
`updateDocument` appears twice, and seems to be identical — copy paste error?

Reply:
Thanks, I was in preparation of the next PR, and forgot to undo. All fixed up @grahamearley